src/Endpoint/EndpointBase.php

```diff
@@ -198,6 +187,7 @@ protected function send(string $method): ResponseInterface
             }
         }
 
+        assert($request instanceof RequestInterface);
```

Comment:
This is added because of backwards functionality. When we lower the dependencies, the `$request->withBody()` method changes the type of `$request` from `RequestInterface` to `MessageInterface`, while both interfaces are implemented in the class. Therefore, PHPStan thinks that `sendRequest($request);` gets the wrong parameter type. Apparently, the union types were not supported yet at the time.

The assert is neutral here, and it is respected by all the quality tools, and PHPStorm typehinting.